test/batch_exchange_utils.js

```diff
@@ -0,0 +1,100 @@
+const MockContract = artifacts.require("MockContract")
+
+const BN = require("bn.js")
+
+const { closeAuction, getBalanceState, getWithdrawableAmount } = require("../scripts/utilities.js")
+
+const { setupGenericStableX } = require("./stablex_utils")
+
+contract("BatchExchange utils", async (accounts) => {
+  describe("getBalanceState()", async () => {
+    it("retrieves balance as in storage", async () => {
+      const batchExchange = await setupGenericStableX()
+      await closeAuction(batchExchange)
+      const erc20 = await MockContract.new()
+      await erc20.givenAnyReturnBool(true)
+
+      await batchExchange.deposit(erc20.address, 0x100)
```

Comment:
Aren't you depositing hex(256) = 0x100 here instead of 100 (0x64)

Reply:
I was also briefly confused but the assertions is with `.toString(16)`. Is there a reason you prefiered to use hex instead of a decimal number for the deposit and assertion?

Reply:
The tested function involves some manual padding with zeros, so I picked a number that would be ambiguous if wrongly padded to the right. It's very unlikely that future changes would involve right padding the output, on the other end the actual number used doesn't really matter for the test, so I went for that. Should I maybe switch to decimal 256?

Reply:
That's fine, maybe add a comment that padding is the reason for this choice. Also, it would be great if we could use the same 0x100 in the assert to avoid confusion. So maybe something like:

`assert.equal(balance.toNumber(), 0x100)`